Junimatic/Patcher.cs

```diff
@@ -0,0 +1,158 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using HarmonyLib;
+using StardewValley;
+
+using static NermNermNerm.Stardew.LocalizeFromSource.SdvLocalize;
+
+namespace NermNermNerm.Junimatic
+{
+    internal static class Patcher
+    {
+        /// <summary>
+        /// Apply the harmony patches for Junimatic
+        /// </summary>
+        internal static void Apply()
+        {
+            ModEntry.Instance.Harmony.Patch(
+                original: AccessTools.Method(typeof(Farmer), nameof(Farmer.addItemToInventoryBool)),
+                prefix: new HarmonyMethod(typeof(Patcher), nameof(Farmer_addItemToInventoryBool_Prefix))
+            );
+            ModEntry.Instance.Harmony.Patch(
+                original: AccessTools.Method(typeof(Game1), nameof(Game1.createItemDebris)),
+                prefix: new HarmonyMethod(typeof(Patcher), nameof(Game1_createItemDebris_Prefix))
+            );
+            ModEntry.Instance.Harmony.Patch(
+                original: AccessTools.Method(typeof(Game1), nameof(Game1.createObjectDebris), [typeof(string), typeof(int), typeof(int), typeof(int), typeof(int), typeof(float), typeof(GameLocation)]),
+                prefix: new HarmonyMethod(typeof(Patcher), nameof(Game1_createObjectDebris_Prefix))
+            );
+        }
+
+        internal static List<Item> InterceptHarvest(Action harvestAction, GameLocation machineLocation)
+        {
+            var playerField = typeof(Game1).GetField(I("_player"), System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Static);
+            var player = Game1.player;
+            try
+            {
+                Patcher.collectedItems = new List<Item>();
+
+                // Set the backing field for Game1.player so that calls to do animation on the farmer don't happen and
+                //  farmer's experience isn't used to calculate the quality of the crops and no xp is gained.
+                playerField!.SetValue(null, Patcher.fakePlayer);
+                Patcher.fakePlayer.currentLocation = machineLocation;
+
+                harvestAction();
+                return Patcher.collectedItems;
+            }
+            finally
+            {
+                playerField!.SetValue(null, player);
+                Patcher.collectedItems = null;
+            }
+        }
+
+
+        private static List<Item>? collectedItems = null;
+        private static Farmer fakePlayer = new FakeFarmer();
```

Comment:
Any concerns about letting the Farmer object persist indefinitely? I know in my implementation I had to call unload because I was instantiating it in a non-persistent class, but I'm wondering if the opposite problem of it persisting indefinitely could be a concern. I wonder if there are some methods that apply to all Farmers in Game1 that could be an issue.

Reply:
I don't know the ramifications either way with any certainty.  I tested it this way and nothing bad happened, that's all I know.

Reply:
Asked the discord and no one there had any concerns so sounds like you're good to go.